packages/leap/lib/src/entities/moving_platform.dart

```diff
@@ -0,0 +1,204 @@
+import 'dart:developer' as developer;
+
+import 'package:flame/extensions.dart';
+import 'package:leap/leap.dart';
+import 'package:tiled/tiled.dart';
+import 'package:tuple/tuple.dart';
+
+abstract class MovingPlatform<T extends LeapGame> extends PhysicalEntity<T> {
+  MovingPlatform({
+    required Vector2 initialPosition,
+    required double tileSize,
+    required this.tilePath,
+    required this.moveSpeed,
+    this.loopMode = MovingPlatformLoopMode.reverseAndLoop,
+    // moving platforms should update before other entities so anything
+    // standing on top of it from the previous frame can be properly moved
+    // with the platform.
+    super.priority = 1,
+  }) : super(static: true, collisionType: CollisionType.standard) {
+    tags.add('ground');
+
+    position = initialPosition;
+
+    // Snap to the closes position on the tile grid
+    position.x = position.x - position.x % tileSize;
+    position.y = position.y - position.y % tileSize;
+
+    this.positionPath = _calculatePositionPath(position, tilePath, tileSize);
+  }
+
+  MovingPlatform.fromTiledObject(
+    TiledObject tiledObject,
+    double tileSize,
+  ) : this(
+          initialPosition: Vector2(tiledObject.x, tiledObject.y),
+          moveSpeed: _parseMoveSpeed(tiledObject),
+          tilePath: _parseTilePath(tiledObject),
+          loopMode: _parseLoopMode(tiledObject),
+          tileSize: tileSize,
+        );
+
+  /// Move speed in tiles per second
+  final Vector2 moveSpeed;
+
+  /// Path of this platform, described as list of nodes of tile offsets.
+  ///
+  /// For example, the tuple {-2, 4} would indicate that the platform should
+  /// move 2 tiles to the left and 4 tiles down from its last node in the
+  /// tile path.
+  ///
+  /// The initial position of the platform is implicitly considered the current
+  /// node when the platform is created.
+  final List<Tuple2<int, int>> tilePath;
```

Comment:
We can probably use records instead? Which is native to the dart language?

Also, regarding this same subject, I implemented an expression language that I am using to define paths in my games, this seems like a good place to use it: https://github.com/erickzanardo/pathxp

The advantages of that expression language over a simple tuple, is that the developer can define if the path should repeat, or if it is infinite, while also defining more complex paths. Let me know what you think

Reply:
Does `pathxp` support diagonal movement? Like `2T2R` for two up and two right in one step?

It does seem like a nearly perfect fit for this usecase

Reply:
It doesn't, but I am happy to implement it